mobile/native/src/trade/order/handler.rs

```diff
@@ -27,16 +27,11 @@ pub async fn submit_order(order: Order) -> Result<()> {
     if let Err(err) = orderbook_client.post_new_order(order.into()).await {
         let order_id = order.id.to_string();
         tracing::error!(order_id, "Failed to post new order. Error: {err:#}");
-        db::update_order_state(order.id, OrderState::Rejected)?;
+        update_order_state(order.id, OrderState::Rejected)?;
         bail!("Could not post order to orderbook");
     }
-    db::update_order_state(order.id, OrderState::Open)?;
 
-    let order = Order {
-        state: OrderState::Open,
-        ..order
-    };
-    ui_update(order);
```

Comment:
❓ Why do we no longer update the UI with the `order` here? I don't think the commit message talks about this change.

Reply:
We do! `update_order_state` handles the UI update, that's why I used it here.

Reply:
I added another commit that might help :)

Reply:
Ah, nice. Thank you. I clearly didn't understand the first version 😛